src/main/java/frc/bionic/swerve/SwerveModule.java

```diff
@@ -0,0 +1,260 @@
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+
+public class SwerveModule
+{
+  // Gear ratio of first two pairs of gears.
+  // Yaw does not depend on the third pair.
+  private double             GEAR_RATIO_12;
+
+  // Gear ratio of all three pairs of gears.
+  // Wheel speed depends on all three.
+  private double             GEAR_RATIO_123;
+
+  // Maximum yaw speed in RPM
+  private double            MAX_YAW_SPEED_RPM;
+
+  // Wheel diameter in meters
+  private double            WHEEL_DIAMETER_METERS;
+
+  // Desired state -- velocity of wheel in RPM, angle in degrees
+  // Angles are measured counter-clockwise, with zero being "robot forward"
+  public double             desiredWheelSpeedRPM;
+  public double             desiredYawDegrees;
+  
+  // Devices, Sensors, Actuators
+  MotorCANSparkMaxNeo       m_driveMotorA;
+  MotorCANSparkMaxNeo       m_driveMotorB;
+  YawEncoderGrayhill63R128  m_yawEncoder;
+  
+  // Shuffleboard-related
+  private String            name;
+  private String            shuffleboardTabName;
+  private NetworkTableEntry sb_yaw;
+  private NetworkTableEntry sb_wheel_rpm;
+  private NetworkTableEntry sb_wheel_calc;
+  private NetworkTableEntry sb_yaw_calc;
+  private NetworkTableEntry sb_trans;
+
+  
+  
+
+
+  public SwerveModule(double gearRatio12, double gearRatio123, double maxYawSpeedRpm,
+                      int driveMotorChannelA, int driveMotorChannelB,
+                      int dioEncoderChannelA, int dioEncoderChannelB,
+                      double wheelDiameterMeters,
+                      String name, String shuffleboardTabName)
+  {
+    // Save arguments
+    GEAR_RATIO_12             = gearRatio12;
+    GEAR_RATIO_123            = gearRatio123;
+    MAX_YAW_SPEED_RPM         = maxYawSpeedRpm;
+    WHEEL_DIAMETER_METERS     = wheelDiameterMeters;
+    this.name                 = name;
+    this.shuffleboardTabName  = shuffleboardTabName;
+
+    // Instantiate the two swerve motors our swerve module incorporates
+    m_driveMotorA =
+      new MotorCANSparkMaxNeo(driveMotorChannelA, name + " A", shuffleboardTabName);
+    m_driveMotorB =
+      new MotorCANSparkMaxNeo(driveMotorChannelB, name + " B", shuffleboardTabName);
+
+    // Instantiate the yaw encoder our swerve module incorporates
+    m_yawEncoder =
+      new YawEncoderGrayhill63R128(dioEncoderChannelA, dioEncoderChannelB, "Grayhill", shuffleboardTabName);
+
+    // If we're using a relative encoder, assume the robot starts up
+    // facing to the zero position. If we're using an absolute
+    // encoder, the `setZero` method is a no-op.
+    m_yawEncoder.setZero();
+
+    // Initialize all of the shuffleboard inputs and outputs
+    initShuffleboard();
+  }
+
+  /**
+   * Reset module's encoder to angle zero
+   */
+  public void setZero()
+  {
+    m_yawEncoder.setZero();
```

Comment:
We will need to carefully consider when to set zero on the lamprey absolute encoders

Reply:
I actually had it as the interface name and then encountered some compilation problems. They may have been due to that VS Code caching problem, though. I'll retry. I agree, it should be the interface name.